src/Servers/Kestrel/Transport.MsQuic/Internal/QuicEnums.cs

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal
+{
+    /// <summary>
+    /// Represents the status of an operation by MSQuic
+    /// </summary>
+    public enum QUIC_STATUS : uint
+    {
+        SUCCESS = 0,
+        PENDING = 459749,
+        CONTINUE = 0x800704DE,
+        OUT_OF_MEMORY = 0x8007000E,
+        INVALID_PARAMETER = 0x80070057,
+        INVALID_STATE = 0x8007139F,
+        NOT_SUPPORTED = 0x80004002,
+        NOT_FOUND = 0x80070490,
+        BUFFER_TOO_SMALL = 0x8007007A,
+        HANDSHAKE_FAILURE = 0x80410000,
+        ABORTED = 0x80004004,
+        ADDRESS_IN_USE = 0x80072740,
+        CONNECTION_TIMEOUT = 0x800704CF,
+        CONNECTION_IDLE = 0x800704D4,
+        INTERNAL_ERROR = 0x80004005,
+        SERVER_BUSY = 0x800704C9,
+        PROTOCOL_ERROR = 0x800704CD,
+        VER_NEG_ERROR = 0x80410001
```

Comment:
These values are platform dependent.

Reply:
Interesting. In some places where this happens we build native shims (one per platform with a consistent public API), in others we unify them in managed code. If we can know for sure that the Linux values are consistent, and the Windows values are consistent, then I'd suggest a managed mapping. Native shims mean different builds for each platform which is a lot of goop for little value :)

Reply:
I'll find the linux values now and add them in. It may be a good idea to run this on linux first before merging.